core/src/main/java/hudson/node_monitors/DfRunner.java

```diff
@@ -0,0 +1,132 @@
+import java.io.InputStreamReader;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+
+class DfRunner {
+	private static final Logger LOGGER = Logger.getLogger(DfRunner.class.getSimpleName());
+
+	private static Map<String, DfCommand> IMPLEMENTATIONS = new LinkedHashMap<String, DfCommand>();
+
+	static {
+		IMPLEMENTATIONS.put("windows", new WindowsDfCommand());
+		IMPLEMENTATIONS.put("linux", new LinuxDfCommand());
+		IMPLEMENTATIONS.put("mac os", new MacOsDfCommand());
+		IMPLEMENTATIONS.put("freebsd", new MacOsDfCommand()); // Same as Mac
+		IMPLEMENTATIONS.put("aix", new AixDfCommand());
+	}
+
+	public String getUsedInodesPercentage() {
+		return findImplementation().get();
+	}
+
+	private DfCommand findImplementation() {
+		String osName = System.getProperty("os.name");
+		for (String key : IMPLEMENTATIONS.keySet()) {
+			if (osName.toLowerCase().startsWith(key)) {
+				LOGGER.info("DfRunner implementation key selected: " + key);
+				return IMPLEMENTATIONS.get(key);
+			}
+		}
+		return new DefaultDfCommand();
+	}
+
+	private static abstract class DfCommand {
+		private String command;
+		public final int line, column;
+
+		DfCommand(String command, int line, int column) {
+			this.command = command;
+			this.line = line;
+			this.column = column;
+		}
+
+		public String get() {
+			try {
+				LOGGER.fine("Inodes monitoring: running '" + command + "' command in " + System.getProperty("user.dir"));
+				Process process = Runtime.getRuntime().exec(command);
+				BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+				for (int i = 1; i < line; ++i) {
+					bufferedReader.readLine(); // Evacuating first lines (header...)
+				}
+				String values = bufferedReader.readLine();
+
+				LOGGER.finest("df values output: " + values);
+				String[] split = values.split(" +");
+				return split[column - 1];
+			}
+			catch (IOException e) {
+				LOGGER.severe("Error while running '" + command + "'");
+				return Messages.InodesMonitor_NotApplicable_OnError();
+			}
+		}
+	}
+
+	private static class WindowsDfCommand extends DfCommand {
+		WindowsDfCommand() {
+			super(null, -1, -1);
+		}
+
+		@Override
+		public String get() {
+			return Messages.InodesMonitor_NotApplicable();
+		}
+	}
+
+	private static class LinuxDfCommand extends DfCommand {
+		LinuxDfCommand() {
+			// The -P can help *not* output the values on two lines when the FS has a long name
+			// But beware the other platform where -P with -i will either
+			// Disable -i (Mac OS) or just fail (AIX)
+			super("df -P -i .", 2, 5);
+		}
+	}
+
+	private static class AixDfCommand extends DfCommand {
+		AixDfCommand() {
+			super("df -i .", 2, 6);
+		}
+	}
+
+	private static class MacOsDfCommand extends DfCommand {
+		MacOsDfCommand() {
+			super("df -i .", 2, 8);
+		}
+	}
+
+	/**
+	 * Tries to run df anyway. Fallback. Will return N/A anyway if an error occurs. Or should we just return N/A directly?
```

Comment:
I would just return N/A
Ideally you could add a popup or whatever else to the table's cell. This popup could mention that Jenkins currently does not support "df"